eng/common/scripts/Verify-RestApiSpecLocation.ps1

```diff
@@ -0,0 +1,281 @@
+          exit 0
+        }
+        $continueValidation = $true
+      }
+    }
+    if($continueValidation -eq $false) {
+      Write-Host "ServiceDir:$ServiceDirectory, no package info is found for package $PackageName, the validation of spec location is ignored."
+      exit 0
+    }
+  }
+  catch {
+    LogError "ServiceDir:$ServiceDirectory, PackageName:$PackageName. Failed to retrieve package and validate package version with exception:`n$_ "
+    exit 1
+  }
+}
+
+try{
+  # Verify package version is not a prerelease version, only continue the validation if the package is GA version
+  Verify-PackageVersion
+
+  $ServiceDir = Join-Path $RepoRoot 'sdk' $ServiceDirectory
+  $PackageDirectory = Join-Path $ServiceDir $PackageName
+  Push-Location $PackageDirectory
+
+  # Load tsp-location.yaml if existed
+  $tspLocationYamlPath = Join-Path $PackageDirectory "tsp-location.yaml"
+  $autorestMdPath = Join-Path $PackageDirectory "src/autorest.md"
+  $swaggerReadmePath = Join-Path $PackageDirectory "swagger/README.md"
+  $tspLocationYaml = @{}
+  if (Test-Path -Path $tspLocationYamlPath) {
+    # typespec scenario
+    $tspLocationYaml = Get-Content -Path $tspLocationYamlPath -Raw | ConvertFrom-Yaml
+    Verify-TspLocation $tspLocationYaml
+  }
+  elseif ($Language -eq "dotnet") {
+    # only dotnet language sdk uses 'autorest.md' to configure the sdk generation
+    if (Test-Path -Path $autorestMdPath) {
+      try {
+        $autorestMdContent = Get-Content -Path $autorestMdPath -Raw
+        Verify-YamlContent $autorestMdContent
+      }
+      catch {
+        Write-Host "ServiceDir:$ServiceDirectory, PackageName:$PackageName. Failed to parse autorest.md file with exception:`n$_ "
+      }
+    }
+  }
+  elseif ($Language -eq "java" -or $Language -eq "js" -or $Language -eq "python" -or $Language -eq "go") {
+    # for these languages we ignore the validation because they always use the latest spec from main branch to release SDK
+    # mgmt plane packages: azure-core-management|azure-resourcemanager|azure-resourcemanager-advisor (java), azure-mgmt-devcenter (python), arm-advisor (js), armaad (go)
+    if($PackageName -match "^(arm|azure-mgmt|azure-resourcemanager|azure-core-management)[-a-z]*$") {
+      Write-Host "ServiceDir:$ServiceDirectory, PackageName:$PackageName. Ignore the validation for $Language management plane package."
+      exit 0
+    }
+    # for these languages they use 'swagger/readme.md' to configure the sdk generation for data plane scenarios
+    if (Test-Path -Path $swaggerReadmePath) {
+      try {
+        $swaggerReadmeContent = Get-Content -Path $swaggerReadmePath -Raw
+        Verify-YamlContent $swaggerReadmeContent
+      }
+      catch {
+        Write-Host "ServiceDir:$ServiceDirectory, PackageName:$PackageName. Failed to parse swagger/readme.md file with exception:`n$_ "
+      }
+    }
+  }
+}
+finally {
+  Pop-Location
+}
```

Comment:
Please add a newline at the end of the file

Reply:
Done.